src/harness/full_activity_dump_helper.py

```diff
@@ -0,0 +1,133 @@
+    # harness is not the fault of the SAS UUT.
+    raise util.TestComponentError('Test Harness Failed. See above stack trace')
+
+
+def _triggerFullActivityDumpAndWaitUntilComplete(sas, sas_admin, ssl_cert,
+                                                 ssl_key):
+  """Triggers generation and loading of dump files from the SAS UUT.
+
+  Will timeout and cause an exception after 7200 seconds (2 hours).
+
+  Args:
+    sas: The SAS to trigger FAD generation of and download the Full Activity
+      Dump Message for.
+    sas_admin: The SasAdminInterface to trigger Full Activity Dump generation
+      on.
+    ssl_cert: ssl certificate to use when making get requests.
+    ssl_key: ssl key to use when making get requests.
+  Returns:
+    Dictionary containing the Full Activity Dump Message from the SAS.
+  Raises:
+    EnvironmentError: If the 7200 second timer is exceeded.
+  """
+  request_time = datetime.utcnow().replace(microsecond=0)
+  sas_admin.TriggerFullActivityDump()
+  signal.signal(signal.SIGALRM,
+                lambda signum,
+                frame: (_ for _ in ()).
+                  throw(EnvironmentError('Full Activity Dump '
+                                         'generation in SAS UUT timed out')))
+  # Timeout after 2 hours if it's not completed.
+  signal.alarm(7200)
+  # Check generation date of full activity dump.
+  while True:
+    dump_message = sas.GetFullActivityDump(ssl_cert, ssl_key)
+    dump_time = datetime.strptime(dump_message['generationDateTime'],
+                                  '%Y-%m-%dT%H:%M:%SZ')
+    if request_time <= dump_time:
+      break
+    time.sleep(10)
+  signal.alarm(0)
+  return dump_message
+
+
+def _processDump(sas, ssl_cert, ssl_key, dump):
+  """Clears any existing dump data and downloads current data.
+
+  Args:
+    sas: A SasInterface object to request
+    ssl_cert: ssl certificate to use when making get requests.
+    ssl_key: ssl key to use when making get requests.
+    dump: The https://base_url/version/dump message response, used to extract
+      the files to be downloaded.
+  Returns:
+    A Full Activity Dump with the FAD data from the given SAS as a dictionary
+    with the fields: cbsd, esc_sensor, zone. Each field is a list of the
+    recordType represented in dictionaries.
+  Raises:
+    AssertError: The FAD response does not match the expected schema or a
+      request does not return 200.
+  """
+  util.assertContainsRequiredFields('FullActivityDump.schema.json', dump)
+  dump_data = {'cbsd': [], 'esc_sensor': [], 'zone': []}
+  for dump_file in dump['files']:
+    if dump_file['recordType'] == 'CoordinationEvent':
+      logging.debug(
+          'CoordinationEvent record skipped in downloading Full Activity Dump')
+      continue
+    util.assertContainsRequiredFields('ActivityDumpFile.schema.json', dump_file)
+    dump_data[dump_file['recordType']].extend(
+        sas.DownloadFile(dump_file['url'], ssl_cert=ssl_cert,
```

Comment:
Do we need to validate the   downloaded file against the schema for the respective record types ?

Reply:
This is entirely possible, however as schema's for all the allowed record types are not yet in master (they will be added by PR #296) could we add this later?